commands/plugin/install.js

```diff
@@ -0,0 +1,197 @@
+      requestManualUpdate(serverlessFilePath);
+      return;
+    }
+
+    const checkIsArrayPluginsObject = (pluginsObject) =>
+      pluginsObject == null || Array.isArray(pluginsObject);
+    // pluginsObject type determined based on the value loaded during the serverless init.
+    if (_.last(serverlessFilePath.split('.')) === 'json') {
+      const serverlessFileObj = await fse.readJson(serverlessFilePath);
+      const newServerlessFileObj = serverlessFileObj;
+      const isArrayPluginsObject = checkIsArrayPluginsObject(newServerlessFileObj.plugins);
+      // null modules property is not supported
+      let plugins = isArrayPluginsObject
+        ? newServerlessFileObj.plugins || []
+        : newServerlessFileObj.plugins.modules;
+
+      if (plugins == null) {
+        throw new ServerlessError(
+          'plugins modules property must be present',
+          'PLUGINS_MODULES_MISSING'
+        );
+      }
+
+      plugins.push(this.options.pluginName);
+      plugins = _.sortedUniq(plugins);
+
+      if (isArrayPluginsObject) {
+        newServerlessFileObj.plugins = plugins;
+      } else {
+        newServerlessFileObj.plugins.modules = plugins;
+      }
+
+      await fse.writeJson(serverlessFilePath, newServerlessFileObj);
+      return;
+    }
+
+    const serverlessFileObj = yaml.load(await fsp.readFile(serverlessFilePath, 'utf8'), {
+      filename: serverlessFilePath,
+      schema: cloudformationSchema,
+    });
+    if (serverlessFileObj.plugins != null) {
+      // Plugins section can be behind veriables, opt-out in such case
+      if (isPlainObject(serverlessFileObj.plugins)) {
+        if (
+          serverlessFileObj.plugins.modules != null &&
+          !Array.isArray(serverlessFileObj.plugins.modules)
+        ) {
+          requestManualUpdate(serverlessFilePath);
+          return;
+        }
+      } else if (!Array.isArray(serverlessFileObj.plugins)) {
+        requestManualUpdate(serverlessFilePath);
+        return;
+      }
+    }
+    await yamlAstParser.addNewArrayItem(
+      serverlessFilePath,
+      checkIsArrayPluginsObject(serverlessFileObj.plugins) ? 'plugins' : 'plugins.modules',
+      this.options.pluginName
+    );
+  }
+
+  async installPeerDependencies() {
+    const pluginPackageJsonFilePath = path.join(
+      this.serviceDir,
+      'node_modules',
+      this.options.pluginName,
+      'package.json'
+    );
+    return fse.readJson(pluginPackageJsonFilePath).then((pluginPackageJson) => {
+      if (pluginPackageJson.peerDependencies) {
+        const pluginsArray = [];
+        Object.entries(pluginPackageJson.peerDependencies).forEach(([k, v]) => {
+          pluginsArray.push(`${k}@"${v}"`);
+        });
+        return Promise.all(pluginsArray.map((plugin) => this.npmInstall(plugin)));
+      }
+      return Promise.resolve();
+    });
+  }
+
+  async npmInstall(name) {
+    return npmCommandDeferred.then((npmCommand) =>
+      execAsync(`${npmCommand} install --save-dev ${name}`, {
```

Comment:
Let's rely on `child-process-ext/spawn` instead (we're slowly migrating to this util)